tests/kernel/mem_protect/userspace/src/inherit.c

```diff
@@ -0,0 +1,105 @@
+/* 
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <kernel_structs.h>
+#include <string.h>
+#include <stdlib.h>
+
+#define INHERIT_STACK_SIZE 1024
+
+/* function prototypes */
+static inline void dummy_start(struct k_timer *timer){ ARG_UNUSED(timer);}
+static inline void dummy_end(struct k_timer *timer){ ARG_UNUSED(timer);}
+
+/* Kernel objects */
+K_THREAD_STACK_DEFINE(test_1_stack, INHERIT_STACK_SIZE);
+K_SEM_DEFINE(inherit_sem, 0, 10);
+K_MUTEX_DEFINE(inhert_mutex);
+K_TIMER_DEFINE(inherti_timer, dummy_start, dummy_end);
+K_MSGQ_DEFINE(inherit_msgq, 10, 10, 2);
+__kernel struct k_thread test_1_tid;
+extern bool valid_fault;
+
+/* variables */
+#ifdef CONFIG_ARM
+u8_t __aligned(32) inherit_buf[32]; /* for mem domain */
+#else
+u8_t __aligned(4096) inherit_buf[4096]; /* for mem domain */
+#endif
+
+K_MEM_PARTITION_DEFINE(inherit_memory_partition, inherit_buf, sizeof(inherit_buf),
+                       K_MEM_PARTITION_P_RW_U_RW);
+
+struct k_mem_partition *inherit_memory_partition_array[] = {
+	&inherit_memory_partition
+};
+
+struct k_mem_domain inherit_mem_domain;
+
+__attribute__((optimize("O0"))) void access_test()
```

Comment:
Any reason to use this __attribute__?  If you're relying on the optimization level for the compiler not to generate code in a specific way, it's better to find another way to accomplish this goal.

Reply:
That was a debug statement. Removed it.